src/librbd/io/ImageRequestWQ.cc

```diff
@@ -311,6 +313,26 @@ void ImageRequestWQ<I>::aio_write(AioCompletion *c, uint64_t off, uint64_t len,
     trace.event("init");
   }
 
+  //block overlapping IOs
+  uint64_t tid = 0;
+  if (native_async && (m_image_ctx.non_blocking_aio || writes_blocked())) {
+    {
+      std::lock_guard locker{ordering_lock};
+      tid = ++m_last_tid;
```

Comment:
Can `m_last_tid` just be an `std::atomic`?

Reply:
Done.